src/Agoda.IoC.NetCore/StartupExtension.cs

```diff
@@ -34,6 +42,21 @@ public static IServiceCollection AutoWireAssembly<T>(this IServiceCollection ser
                     attr => new RegistrationContext(attr, x.ToType, isMockMode)
                 ))
                 .ToList();
+
+            if (registrations.Any(x => !x.Validation.IsValid))
+            {
+                var exceptions  = new ContainerRegistrationContextException("There are validations errors!!!");
+                exceptions
+                    .RegistrationContextExceptions
+                    .AddRange(registrations.Where(x => !x.Validation.IsValid)
+                                           .Select(ex => new RegistrationContextException(ex, ex.Validation.ErrorMessage)));
```

Comment:
The naming `ex` is a bit misleading. You are selecting the registration context, not an exception.